common/src/main/java/tachyon/security/authentication/CustomAuthenticationProviderImpl.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.security.authentication;
+
+import javax.security.sasl.AuthenticationException;
+
+import com.google.common.base.Strings;
+
+import tachyon.Constants;
+import tachyon.conf.TachyonConf;
+import tachyon.util.CommonUtils;
+
+public class CustomAuthenticationProviderImpl implements AuthenticationProvider {
+
+  private final AuthenticationProvider mCustomProvider;
+
+  public CustomAuthenticationProviderImpl() {
+    TachyonConf conf = new TachyonConf();
```

Comment:
NIT: in the current code base, we are trying to avoid creating new `TachyonConf` object inside different classes but let their constructors take a reference of  a `TachyonConf` object as a parameter. The purpose of making this effort is simply to avoid having duplicated copy of the same object and also saving the setup overhead. Could you make change to your constructors of `CustomAuthenticationProviderImpl` and also maybe adjust tests accordingly.